src/node/desktop/src/main/detect_r.ts

```diff
@@ -13,22 +13,27 @@
  *
  */
 
-import { app, dialog } from 'electron';
-import { exec } from 'child_process';
-import { promisify } from 'util';
+import { dialog } from 'electron';
+import { execSync } from 'child_process';
 import { existsSync } from 'fs';
-import { assert } from 'console';
 
 import { logger } from '../core/logger';
 import { Environment, getenv, setVars } from '../core/environment';
-import { FilePath } from '../core/file-path';
 
-import { appState } from './app-state';
+import { dirname } from 'path';
 
-const asyncExec = promisify(exec);
+interface REnvironment {
+  rScriptPath: string,
+  version: string,
+  envVars: Environment,
+}
+
+function rNotFoundErrorMessage(): string {
```

Comment:
See comment later on not throwing a string directly, either use `throw Error(rNotFoundErrorMessage())` or change this function to do that for you and return an `Error`.